automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainMultiGroupApiSpec.scala

```diff
@@ -0,0 +1,417 @@
+package org.broadinstitute.dsde.test.api
+
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+import org.broadinstitute.dsde.workbench.config.{Credentials, UserPool}
+import org.broadinstitute.dsde.workbench.fixture.{BillingFixtures, GroupFixtures, WorkspaceFixtures}
+import org.broadinstitute.dsde.workbench.service.BillingProject.BillingProjectRole
+import org.broadinstitute.dsde.workbench.service.{AclEntry, Orchestration, Rawls, WorkspaceAccessLevel}
+import org.scalatest._
+import spray.json.{JsValue, JsonParser}
+
+import scala.util.Try
+
+
+class AuthDomainMultiGroupApiSpec extends FreeSpec with Matchers with WorkspaceFixtures with BillingFixtures with GroupFixtures {
+
+  /*
+   * Unless otherwise declared, this auth token will be used for API calls.
+   * We are using a curator to prevent collisions with users in tests (who are Students and AuthDomainUsers), not
+   *  because we specifically need a curator.
+   */
+
+  val defaultUser: Credentials = UserPool.chooseCurator
+  val authTokenDefault: AuthToken = defaultUser.makeAuthToken()
+
+
+  "A workspace" - {
+    "with multiple groups inside of it" - {
+
+      "can be created" in {
+
+        val user = UserPool.chooseAuthDomainUser
+        implicit val authToken: AuthToken = authTokenDefault
+
+        withGroup("AuthDomainOne", List(user.email)) { groupOne =>
+          withGroup("AuthDomainTwo", List(user.email)) { groupTwo =>
+
+            withCleanBillingProject(user) { projectName =>
+              withWorkspace(projectName, "GroupsApiSpec_workspace", Set(groupOne, groupTwo)) { workspaceName =>
+
+                // user is in members emails in two authdomain groups
+                groupNameToMembersEmails(groupOne) should contain allElementsOf List(user.email)
+                groupNameToMembersEmails(groupTwo) should contain allElementsOf List(user.email)
```

Comment:
for these you should just be able to do something like `... should contain (user.email)` instead of checking all elements of a 1 element list.

Reply:
fixed